### PR TITLE
Fix SLES 12 Missing Dependency by using libhiredis0.13 

### PR DESCRIPTION
**Pull request by sophieyfang**
Install libhiredis.so.0.13 in the build container as SLES 12 only has 0.13

### CODE REVIEW

File: build/x86_64_sles12/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 FROM opensuse/leap:42.3
 
-RUN zypper addrepo https://download.opensuse.org/repositories/devel:libraries:c_c++/SLE_12_SP3/devel:libraries:c_c++.repo \
+RUN zypper addrepo https://download.opensuse.org/repositories/Apache/SLE_12_SP4/Apache.repo \
+ && zypper addrepo https://download.opensuse.org/repositories/server:database/SLE_15/server:database.repo \
```
Comment (igorpeshansky) [inline build/x86_64_sles12/Dockerfile]: Can you please add a comment explaining why we're adding a SLES 15 repo here?

Comment (sophieyfang) [reply build/x86_64_sles12/Dockerfile]: Done

Comment (igorpeshansky) [reply build/x86_64_sles12/Dockerfile]: Actually, I suspect we don't really need to install `libhiredis0_13` explicitly — we will pick up `hiredis-devel` version 0.13 from this repo, which will pull in the right version of the shared library. Can we verify that and, if so, update the comment? Thanks.

Comment (sophieyfang) [reply build/x86_64_sles12/Dockerfile]: You are right. I removed the line for installing the  libhiredis0_13  and it still has the right version. I have updated the code and I will update the comment too
8e3f149c42e5:/ # zypper search libhiredis    
Loading repository data...
Warning: Repository 'NON OSS Update' appears to be outdated. Consider using a different mirror or server.
Warning: Repository 'OSS Update' appears to be outdated. Consider using a different mirror or server.
Reading installed packages...

S | Name                     | Summary                                      | Type   
--+--------------------------+----------------------------------------------+--------
  | libhiredis0_12           | Shared library for hiredis                   | package
i | libhiredis0_13           | Shared library for hiredis                   | package
  | libhiredis0_13-debuginfo | Debug information for package libhiredis0_13 | package

Comment (sophieyfang) [reply build/x86_64_sles12/Dockerfile]: Comment just updated.

**Comment by qingling128** [comment]
Hmm, is this one ready to merge?

**Comment by sophieyfang** [comment]
Wait for igor to review. I have made changes accordingly and confirmed it works

**Comment by qingling128** [comment]
Oh, so the image already has this fix. Cool cool.

File: build/x86_64_sles12/Dockerfile [hunk 2]
```diff
@@ -28,9 +30,9 @@ RUN zypper addrepo https://download.opensuse.org/repositories/devel:libraries:c_
         varnish-devel \
         wget \
         which \
- && wget https://ftp.gwdg.de/pub/opensuse/repositories/home%3A/Herbster0815/openSUSE_Leap_15.1/noarch/automake-1.16.1-lp151.92.1.noarch.rpm \
- && rpm --force -Uvh automake-1.16.1-lp151.92.1.noarch.rpm \
- && rm automake-1.16.1-lp151.92.1.noarch.rpm \
+ && wget https://ftp.gwdg.de/pub/opensuse/repositories/home:/Herbster0815/openSUSE_Leap_15.1/noarch/automake-1.16.1-lp151.93.1.noarch.rpm \
+ && rpm --force -Uvh automake-1.16.1-lp151.93.1.noarch.rpm \
+ && rm automake-1.16.1-lp151.93.1.noarch.rpm \
```
Comment (igorpeshansky) [inline build/x86_64_sles12/Dockerfile]: [Optional] We really ought to do something about having to update this all the time, e.g., change to:
```
 && zypper addrepo https://ftp.gwdg.de/pub/opensuse/repositories/home:/Herbster0815/openSUSE_Leap_15.1/home:Herbster0815.repo \
 && wget -O Herbster0815.key https://download.opensuse.org/repositories/home:/Herbster0815/openSUSE_Leap_15.1/repodata/repomd.xml.key \
 && rpm --import Herbster0815.key \
 && rm -f Herbster0815.key \
 && zypper -n install home_Herbster0815:automake \
```
But that's probably material for a separate PR.

**Pull request merged**